java/daikon/chicory/Instrument24.java

```diff
@@ -0,0 +1,1757 @@
+  }
+
+  /**
+   * Output the given method with no changes.
+   *
+   * @param classBuilder for the given method
+   * @param mm MethodModel for the given method
+   * @param mgen describes the given method
+   */
+  private void outputMethodUnchanged(ClassBuilder classBuilder, MethodModel mm, MethodGen24 mgen) {
+    classBuilder.withMethod(
+        mm.methodName().stringValue(),
+        mm.methodTypeSymbol(),
+        mm.flags().flagsMask(),
+        methodBuilder -> copyMethod(methodBuilder, mm, mgen));
+  }
+
+  /**
+   * Copy the given method with no changes.
+   *
+   * @param methodBuilder for the given method
+   * @param methodModel for the given method
+   * @param mgen describes the given method
+   */
+  private void copyMethod(MethodBuilder methodBuilder, MethodModel methodModel, MethodGen24 mgen) {
+
+    for (MethodElement me : methodModel) {
+      debugInstrument.log("MethodElement: %s%n", me);
+      switch (me) {
+        case CodeModel codeModel ->
+            methodBuilder.withCode(codeBuilder -> copyCode(codeBuilder, mgen.getInstructionList()));
+
+          // copy all other MethodElements to output class (unchanged)
+        default -> methodBuilder.with(me);
+      }
+    }
+  }
+
+  /**
+   * Copy code for the given method with no changes.
+   *
+   * @param codeBuilder for the given method's code
+   * @param instructions instruction list for method
+   */
+  private void copyCode(CodeBuilder codeBuilder, List<CodeElement> instructions) {
+
+    for (CodeElement ce : instructions) {
+      debugInstrument.log("CodeElement: %s%n", ce);
+      codeBuilder.with(ce);
+    }
+  }
+
+  /**
+   * Instrument the given method using {@link #instrumentCode}.
+   *
+   * @param methodBuilder for the given method
+   * @param methodModel for the given method
+   * @param mgen describes the given method
+   * @param curMethodInfo provides additional information about the method
+   */
+  private void instrumentMethod(
+      MethodBuilder methodBuilder,
+      MethodModel methodModel,
+      MethodGen24 mgen,
+      MethodInfo curMethodInfo,
+      int method_info_index) {
+
+    for (MethodElement me : methodModel) {
+      debugInstrument.log("MethodElement: %s%n", me);
+      switch (me) {
+        case CodeModel codeModel ->
+            methodBuilder.withCode(
+                codeBuilder ->
+                    instrumentCode(codeBuilder, codeModel, mgen, curMethodInfo, method_info_index));
+
+          // copy all other MethodElements to output class (unchanged)
+        default -> methodBuilder.with(me);
+      }
+    }
+  }
+
+  /**
+   * Insert the our instrumentation code into the instruction list for the given method.
+   *
+   * @param instructions instruction list for method
+   * @param mgen describes the given method
+   * @param curMethodInfo provides additional information about the method
+   * @param minfo for the given method's code
+   */
+  private void insertInstrumentation(
```

Comment:
What is the relationship between `insertInstrumentation` and `instrumentCode`?  Why are their names so different, when they seem to have very similar specifications?